src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -581,3 +582,43 @@ def finalize(self) -> InheritedContext:
             return self
         tree = self._tree.renumber(-1)
         return InheritedContext(tree, self.found)
+
+
+T = TypeVar("T")
+
+
+class DependencyGraphWalker(abc.ABC, Generic[T]):
+
+    def __init__(self, graph: DependencyGraph, walked_paths: set[Path] | None, path_lookup: PathLookup):
+        self._graph = graph
+        self._walked_paths: set[Path] = set() if walked_paths is None else walked_paths
```

Comment:
No because the motivation for providing a set is to use it from outside (notably during testing). An empty `walked_paths` set returns False, thus breaking the contract by creating a new set